zds/member/tests/tests_views.py

```diff
@@ -1723,3 +1724,100 @@ def send_messages(messages):
+        self.assertContains(response, self.user_ipv4_same_ip_2.user.username)
+        self.assertContains(response, self.user_ipv4_same_ip_1.last_ip_address)
+        self.assertNotContains(response, self.user_ipv4_different_ip.user.username)
+
+    def test_different_ipv4(self) -> None:
+        self.client.force_login(self.staff)
+        response = self.client.get(reverse(member_from_ip, args=[self.user_ipv4_different_ip.last_ip_address]))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "member/admin/memberip.html")
+        self.assertContains(response, self.user_ipv4_different_ip.user.username)
+        self.assertContains(response, self.user_ipv4_different_ip.last_ip_address)
+        self.assertNotContains(response, self.user_ipv6_same_ip_1.user.username)
+
+    def test_same_ipv6_and_same_ipv6_network(self) -> None:
+        self.client.force_login(self.staff)
+        response = self.client.get(reverse(member_from_ip, args=[self.user_ipv6_same_ip_1.last_ip_address]))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "member/admin/memberip.html")
+        self.assertContains(response, self.user_ipv6_same_ip_1.user.username)
+        self.assertContains(response, self.user_ipv6_same_ip_2.user.username)
+        self.assertContains(response, self.user_ipv6_same_network.user.username)
+        self.assertNotContains(response, self.user_ipv6_different_network.user.username)
+
+    def test_same_ipv6_network_but_different_ip(self) -> None:
+        self.client.force_login(self.staff)
+        response = self.client.get(reverse(member_from_ip, args=[self.user_ipv6_same_network.last_ip_address]))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "member/admin/memberip.html")
+        self.assertContains(response, self.user_ipv6_same_network.user.username)
+        self.assertContains(response, self.user_ipv6_same_ip_1.user.username)
+        self.assertContains(response, self.user_ipv6_same_ip_2.user.username)
+        self.assertNotContains(response, self.user_ipv6_different_network.user.username)
+
+    def test_different_ipv6_network(self) -> None:
+        self.client.force_login(self.staff)
+        response = self.client.get(reverse(member_from_ip, args=[self.user_ipv6_different_network.last_ip_address]))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "member/admin/memberip.html")
+        self.assertContains(response, self.user_ipv6_different_network.user.username)
+        self.assertNotContains(response, self.user_ipv6_same_ip_1.user.username)
+        self.assertNotContains(response, self.user_ipv6_same_ip_2.user.username)
+        self.assertNotContains(response, self.user_ipv6_same_network.user.username)
+
+    def test_access_rights_to_ip_page(self) -> None:
+        regular_user = ProfileFactory()
+        regular_user.user.save()
```

Comment:
J'aurai tendance à mettre ces lignes là dans `setUp` pour garder la cohérence avec les autres tests.

Reply:
C'est ce que j'avais pensé de base, mais ils ne sont nécessaires que pour le dernier, alors je m'étais dit que c'était un peu plus opti de les mettre dans le seul qui en avait besoin. Pas nécessaire, probablement ?

Reply:
Quoi qu'il en soit, c'est vraiment du détail. J'ai le sentiment que ça dépend ce que tu veux privilégier.

1. Si tu veux privilégier l'homogénéité globale de ta classe, tu le mets dans le setup.
2. Si tu veux mettre les définitions au plus près de l'usage, tu le mets là où il est actuellement.

Dans les deux cas, ça reste anecdotique d'un point de vue performance. Surtout que les tests sont joués qu'une fois de temps en temps. :)